### PR TITLE
Added task to disable default installed apps

### PR DESCRIPTION
**Pull request by jspaans91**
Added a task to disable apps that are enabled by default and you maybe not want. The variable for the list with apps you want to disable is "nextcloud_disable_apps"

### CODE REVIEW

**Comment by jspaans91** [comment]
@aalaesar sorry for bothering you, but can you check if we want this functionality be added to this role or not?

If not, I'll add it to my own Nexcloud customization role.

Cc @staticdev

**Comment by aalaesar** [comment]
> @aalaesar sorry for bothering you, but can you check if we want this functionality be added to this role or not?
> 
> If not, I'll add it to my own Nexcloud customization role.
> 
> Cc @staticdev

hello @jspaans91  I was reviewing the code at the moment

> @aalaesar sorry for bothering you, but can you check if we want this functionality be added to this role or not?
> 
> If not, I'll add it to my own Nexcloud customization role.
> 
> Cc @staticdev

hello @jspaans91  and thank you for the idea and the contribution
I was reviewing the PR ealier and I think your idea is great, as nextcloud is coming with more and more shipped application. One may not want/need them all.

As the role is a little big, I tend to look for minimising the number of tasks to do things. 🤔 
The two main tasks to do it are here, there is also some refinement required to simplify how to use the list of apps returned by the occ cli without using tasks.
I can make suggestions if you're ok 😃

File: tasks/main.yml [hunk 1]
```diff
@@ -106,6 +106,33 @@
     - nextcloud_apps is mapping
   tags: install_apps
 
+- name: Disable Nextcloud Apps
```
Comment (staticdev) [inline tasks/main.yml:109]: I think it is not needed to add this session to `main.yml`. A more compact and easy to maintain solution would be to add to the end of `nc_installation.yml` a block that executes `when: nextcloud_disable_apps is defined` to run `php occ app:disable "{{ nc_app_name }}"` with items from this list.

File: defaults/main.yml [hunk 2]
```diff
@@ -105,6 +105,7 @@ nextcloud_tls_session_cache_size: 50m  # in Byte or human readable size notation
 
 # [APPS]
 nextcloud_apps: {}
+nextcloud_disable_apps: {}
```
Comment (staticdev) [inline defaults/main.yml:108]: When adding a new parameter, especially when default is empty, is very good if we have an actual example with a list in the [Example Playbooks](https://github.com/aalaesar/install_nextcloud#example-playbook).

File: tasks/nc_disable_apps.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (staticdev) [inline tasks/nc_disable_apps.yml:1]: With the comment from `main.yml` this makes also this file not necessary.

**Review by staticdev — CHANGES_REQUESTED** [review]

**Comment by staticdev** [comment]
Thanks a lot for the contribution again @jspaans91! I also like the feature. I made some suggestions already, not sure if would be in line with @aalaesar was also thinking. Feel free to comment or propose alternatives.

File: tasks/main.yml (hunk 1 above)
Comment (aalaesar) [reply tasks/main.yml:109]: @staticdev suggestion is even simpler than what I had i mind, so I :+1:  it

1tasks looping on the list and changed status depending of the stdout:
- ok : "No such app enabled: {{ item }}"
- changed: "{{ item }} <a version> enabled"

rc is always 0 in both case

should do something like that:
```yaml
changed_when:
  - "'enabled' in stdout"
  - "'No such app" not in stdout
```

**Comment by staticdev** [comment]
Hoi @jspaans91 just a vrindelijke ping. Did you had opportunity to take a look at our code review? What do you think about it?

**Comment by staticdev** [comment]
@jspaans91 I will closed this PR for inactivity, if you are able to reply us I will be happy to reopen.

**Pull request closed without merging**